server/models/k8s_components_registration.go

```diff
@@ -49,51 +49,61 @@ type K8sRegistrationFunction func(ctxt context.Context, config []byte, ctxID str
 
 // start registration of components for the contexts
 func (cg *ComponentsRegistrationHelper) RegisterComponents(ctxs []*K8sContext, regFunc []K8sRegistrationFunction, eb *events.EventStreamer, reg *meshmodel.RegistryManager) {
+	if viper.GetBool("SKIP_COMP_GEN") {
+		return
+	}
+
 	for _, ctx := range ctxs {
-		ctxName := ctx.Name
 		ctxID := ctx.ID
+
 		// do not do anything about the contexts that are not present in the ctxRegStatusMap
-		if status, ok := cg.ctxRegStatusMap[ctxID]; ok {
-			if !viper.GetBool("SKIP_COMP_GEN") {
-				// only start registering components for contexts whose status is NotRegistered
-				if status == NotRegistered {
-					id, _ := uuid.NewV4()
-					// update the status
-					cg.ctxRegStatusMap[ctxID] = Registering
-					cg.log.Info("Registration of ", ctxName, " components started for contextID: ", ctxID)
-					req := meshes.EventsResponse{
-						Component:     "core",
-						ComponentName: "Kubernetes",
-						EventType:     meshes.EventType_INFO,
-						Summary:       fmt.Sprintf("Registration for Kubernetes context \"%s\" started", ctxName),
-						Details:       fmt.Sprintf("Registration for Kubernetes context \"%s\" started with context ID %s", ctxName, ctxID),
-						OperationId:   id.String(),
-					}
-					eb.Publish(&req)
-					go func(ctx *K8sContext) {
-						// set the status to RegistrationComplete
-						defer func() {
-							cg.ctxRegStatusMap[ctxID] = RegistrationComplete
-
-							cg.log.Info(ctxName, " components for contextID:", ctxID, " registered")
-						}()
-
-						// start registration
-						cfg, err := ctx.GenerateKubeConfig()
-						if err != nil {
-							cg.log.Error(err)
-							return
-						}
-						for _, f := range regFunc {
-							err = f(context.Background(), cfg, ctxID, reg, eb, ctxName)
-							if err != nil {
-								cg.log.Error(err)
-								return
-							}
-						}
-					}(ctx)
+		// only start registering components for contexts whose status is NotRegistered
+		status, ok := cg.ctxRegStatusMap[ctxID]
+		if !ok || status != NotRegistered {
+			continue
+		}
+
+		ctxName := ctx.Name
+		id, _ := uuid.NewV4()
+
+		// update the status
+		cg.ctxRegStatusMap[ctxID] = Registering
+		cg.log.Info("Registration of ", ctxName, " components started for contextID: ", ctxID)
+		req := meshes.EventsResponse{
+			Component:     "core",
+			ComponentName: "Kubernetes",
+			EventType:     meshes.EventType_INFO,
+			Summary:       "Registration for kubernetes context " + ctxName + " started",
+			Details:       "Registration of " + ctxName + " components started for contextID: " + ctxID,
+			OperationId:   id.String(),
```

Comment:
Thanks, @Azanul!

Will you please review the recent changes to the related message sent upon start/end of Kubernetes component registration? That new verbiage can be used here.